src/main/resources/default/assets/common/core.js.pasta

```diff
@@ -381,6 +381,12 @@ sirius.addElementVisibleListener = function (selector, listener, distanceFactor)
         const factor = distanceFactor || 1;
         const rect = element.getBoundingClientRect();
 
+        // Simple visibility check to ignore hidden elements
+        if (rect.height === 0 || rect.width === 0) {
```

Comment:
**Optional:** Wouldn't [checkVisibility](https://developer.mozilla.org/en-US/docs/Web/API/Element/checkVisibility) be a saver approach for modern browsers?
But I am also fine with the current fix until we see a false-positive (a zero size element that isn't actually hidden).

Reply:
Would be possible if a function like `sirius.isElementVisible` (or something like that) is added with something like this:

```js
sirius.isElementVisible = function (_element) {
	if (_element.checkVisibility) {
		return _element.checkVisibility();
	} else {
		const rect = _element.getBoundingClientRect();
		return rect.height > 0 || rect.width > 0;
	}
}
```

Not sure if `||` or `&&` would be better in the last condition